### PR TITLE
Add CLI tool scaffold to replace old nodejs SKR tests

### PR DESCRIPTION
**Pull request by MarekMichali**
**Description**

Add scaffolding for a CLI to replace SKR tests written in nodejs.

Changes proposed in this pull request:

- Add provisioning command
- Add deprovisioning command
- Add scaffolding for other KEB operations such as update, binding creation, etc.

**Related issue(s)**

See #1588

### CODE REVIEW

File: testing/e2e/skr-tester/pkg/logger/logger.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package logger
+
+import (
+	"github.com/int128/kubelogin/pkg/infrastructure/logger"
+	"github.com/sirupsen/logrus"
```
Comment (KsaweryZietara) [inline testing/e2e/skr-tester/pkg/logger/logger.go:5]: Is it possible to use `slog` instead?

Comment (MarekMichali) [reply testing/e2e/skr-tester/pkg/logger/logger.go:5]: I will change it in next iteration.

File: testing/e2e/skr-tester/cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	"skr-tester/pkg/command"
+	"syscall"
+)
+
+func main() {
+	setupCloseHandler()
+	cmd := command.New()
+
+	err := cmd.Execute()
+	if err != nil {
```
Comment (jaroslaw-pieszka) [inline testing/e2e/skr-tester/cmd/main.go:16]: As for YAGNI principle, we could leave this `1` assuming that no elaborate shell scripts will call this CLI tool.

File: testing/e2e/skr-tester/pkg/command/provision.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		fmt.Printf("Error provisioning SKR: %v\n", err)
+	} else {
+		fmt.Printf("Provision operation ID: %s\n", resp["operation"].(string))
+	}
+
+	return nil
+}
+
+func (cmd *ProvisionCommand) Validate() error {
+	if cmd.planID != "" || cmd.region != "" {
```
Comment (jaroslaw-pieszka) [inline testing/e2e/skr-tester/pkg/command/provision.go:63]: not consistent with the error message

File: testing/e2e/skr-tester/pkg/keb/client.go [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+)
+
+const (
+	SCOPES                     = "broker:write"
```
Comment (jaroslaw-pieszka) [inline testing/e2e/skr-tester/pkg/keb/client.go:16]: It is just one scope. Not sure how to name it.

File: testing/e2e/skr-tester/pkg/keb/client.go [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+	KYMA_SERVICE_ID            = "47c9dcbf-ff30-448e-ab36-d3bad66ba281"
+	trialPlanID                = "7d55d31d-35ae-4438-bf13-6ffdfa107d9f"
+	DEFAULT_EXPIRATION_SECONDS = 600
```
Comment (jaroslaw-pieszka) [inline testing/e2e/skr-tester/pkg/keb/client.go:19]: inconsistent naming convention: constants with caps or snakeCase?

File: testing/e2e/skr-tester/pkg/keb/client.go [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusAccepted {
+		return nil, fmt.Errorf("error calling KEB: %s %s", resp.Status, resp.Status)
+	}
+
+	var result map[string]interface{}
+	json.NewDecoder(resp.Body).Decode(&result)
+	return result, nil
+}
+
+func (c *KEBClient) CallKEBWithoutToken(payload interface{}, endpoint, verb string) error {
```
Comment (jaroslaw-pieszka) [inline testing/e2e/skr-tester/pkg/keb/client.go:175]: Should we use `KEB`? I am hesitant.
Then we have such awkward names (given the snakeCase convention).
Full name would be even more awkward `CallKymaEnvironmentBrokerWithoutToken`
What about `Broker`?

File: testing/e2e/skr-tester/pkg/keb/client.go [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+	fmt.Println("Response:", string(body))
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusForbidden && resp.StatusCode != http.StatusUnauthorized {
+		return fmt.Errorf("unexpected status: %d", resp.StatusCode)
+	}
+	return nil
+}
+
+func (c *KEBClient) GetSKR(instanceID string) (map[string]interface{}, error) {
```
Comment (jaroslaw-pieszka) [inline testing/e2e/skr-tester/pkg/keb/client.go:203]: Like `KEB` `SKR` is a name we use on daily basis but still it is kind of cryptic.
I do not know if there is better option. Could we try to invent something?

File: testing/e2e/skr-tester/cmd/main.go (hunk 2 above)
Comment (MarekMichali) [reply testing/e2e/skr-tester/cmd/main.go:16]: I don't foresee a complex script using this tool. The script will only execute commands from this tool.

File: testing/e2e/skr-tester/pkg/command/provision.go (hunk 3 above)
Comment (MarekMichali) [reply testing/e2e/skr-tester/pkg/command/provision.go:63]: Fixed to &&.

Comment (MarekMichali) [reply testing/e2e/skr-tester/pkg/command/provision.go]: Fixed to &&.

File: testing/e2e/skr-tester/pkg/keb/client.go (hunk 4 above)
Comment (MarekMichali) [reply testing/e2e/skr-tester/pkg/keb/client.go]: Changed to `scope`.

File: testing/e2e/skr-tester/pkg/keb/client.go (hunk 5 above)
Comment (MarekMichali) [reply testing/e2e/skr-tester/pkg/keb/client.go]: Changed to snakeCase.

File: testing/e2e/skr-tester/pkg/keb/client.go (hunk 6 above)
Comment (MarekMichali) [reply testing/e2e/skr-tester/pkg/keb/client.go:175]: We can discuss it with the team today.

File: testing/e2e/skr-tester/pkg/keb/client.go (hunk 7 above)
Comment (MarekMichali) [reply testing/e2e/skr-tester/pkg/keb/client.go:203]: We can discuss it with the team today.

File: testing/e2e/skr-tester/pkg/keb/client.go (hunk 6 above)
Comment (MarekMichali) [reply testing/e2e/skr-tester/pkg/keb/client.go]: We decided to change KEB to broker and SKR to instance.

File: testing/e2e/skr-tester/pkg/keb/client.go (hunk 7 above)
Comment (MarekMichali) [reply testing/e2e/skr-tester/pkg/keb/client.go]: We decided to change KEB to broker and SKR to instance.

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**